CefSharp.Core/Internals/CefRegisterCdmCallbackWrapper.h

```diff
@@ -0,0 +1,65 @@
+// Copyright � 2010-2016 The CefSharp Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#pragma once
+
+#include "Stdafx.h"
+#include "CefWrapper.h"
+
+namespace CefSharp
+{
+    namespace Internals
+    {
+        private class CefRegisterCdmCallbackWrapper : public CefRegisterCdmCallback
+        {
+        private:
+            gcroot<TaskCompletionSource<CdmRegistration^>^> _taskCompletionSource;
```

Comment:
> Never mind I get it, just replace the _taskCompletionSource with it. :)

Pretty much, see https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Core/Internals/CefCompletionCallbackAdapter.h for an example.

Also `IRegisterCdmCallback` should implement `IDisposable`